### ISSUE TITLE
Angularjs specific question about how to use RestFul filters

### ISSUE THREAD

**Issue by doodirock** [RESTful-Drupal/restful#769]
I've been building a small app with the api and trying to find some robust examples of things like how to post, update, login etc,  All has been going decently well until tonight when I ran into an issue with filters.  I hate to use the issue queue for Angular based questions, but i figure since a lot of you guys are using RestFul with Angular you might have an answer. 

In my example below I have a drop down that takes the ng-model of the selection and applies this to the list of paramenters and then fires a method to filter my list.  However I'm not sure how to handle the array style query that a filter uses with ng-resource.  

```
$scope.paramers = {
    include: 'playing', 
    sort: '-id'
};
$scope.refresh = function () {
    LFGFactory.query($scope.paramers, function success (response) {
        $scope.loading = true;
        var data = response.data;
        if (data.length >= 1) {
            $scope.rowList = data;
            $scope.loading = false;
        } else {
            $scope.loading = false;
        }
    },
    function err (data) {
        console.log(data);
    });
};
```

My view below 

```
    <div class="form-group pull-right">
        <select id="plat-sel" name="plat-sel" class="form-control" ng-model="paramers.filter" ng-change="refresh()">
            <option value="" disabled selected>Filter by Platform</option>
            <option value="1183">Xbox One</option>
            <option value="1184">PlayStation 4</option>
            <option value="1182">PC</option>
            <option value="1188">Wii U</option>
            <option value="1186">Xbox 360</option>
            <option value="1185">PlayStation 3</option>
        </select>
    </div>
```

Currently $scope.paramers is sending www.site.com/api/lfg?filter=1184 back to drupal, but I'm not sure how to pass the required style of www.site.com/api/lfg?filter[id]=1184.  How have you all handle this type of request in Angular using $resource?  Couldnt seem to find anything in the example app.

SIDE NOTE:  Are there any docs for the proper payload in terms of how to post or update to entitys?

**Comment by jeff-h** [comment]
Sorry, can't help you with your Angular question, but re: posting / updating an entity, does the following from the wiki help? (it was written for 1.x — not sure whether it will need slight adjusting for 2.x)

https://github.com/RESTful-Drupal/restful/wiki/Saving-an-entity-back-to-Drupal

Also, it would be awesome if you could put some of your Angular knowledge into the wiki. I'm using Backbone so the wiki will have that covered as best I can, but as you say, a lot of people are into Angular and it would be great to have that also covered in the wiki.

**Comment by jeff-h** [comment]
ps please don't be shy re: the wiki... if you now know something that you wished was in the wiki before you knew what you now know, put it in lol Even if just as a reference for yourself later on.

**Comment by doodirock** [comment]
Will do Jeff.  Once i get a better grasp on things ill add whatever i find out in the wiki.  Hopefully someone here has worked with the filters in angular

**Comment by wonder95** [comment]
@doodirock, can't you take a look at what you're getting from ng-model inside of your refresh() function in the debugger? It's hard to see how your URL is being created without the LFGFactory.query service code.

**Comment by e0ipso** [comment]
I only have [limited angular experience](https://github.com/e0ipso/message-center) but I'll be paying attention to the debate here!

**Comment by doodirock** [comment]
@wonder95 yep, as you can see the model is just being updated but the value of the select and being placed into the paramers object like so:

```
$scope.paramers = {
    include: 'playing', 
    sort: '-id',
};  filter: 'value'
```

Value in this context is just the ID of the taxonomy term that I'm trying to filter by.  The really issue resides in how I send this off in the get request in the proper format (IE filter[id]=id) since it current is sent off as filter=id.

Here is my service for more info.

```
  .factory('LFGFactory', function($resource) {

    var base = 'http://example.com/api/v1.0/';

    return $resource(base +'lfg', {},
        {
          update: {
            method: 'PUT',
            isArray: true
          },
          delete: {
            method: 'DELETE',
            isArray: true
          },
          query: {
            method: 'GET',
            isArray: false
          }
        }
    );
  }); 
```

Maybe Restful accepts another style instead of using array parameters.  The odd thing is im sure other apis use this style of params and yet I don't see much on stack about the subject.  I'm not even sure how to format the values within the object

**Comment by doodirock** [comment]
Just an update.  Still haven't found any solution to using filters from within my angular app.  Seems posting to the server in this style is a huge pain in the butt.  Oddly stack and IRC haven't been helpful either.  Surprised at how rare this type of posting is required by other APIs.  Seems pretty normal to me

**Comment by e0ipso** [comment]
I am not sure what you want to accomplish. Can you translate your request to a cURL command? Would that be working then?

**Comment by doodirock** [comment]
I'm just trying to figure out a way to sort via filter within Angular.  Can't find a way to hit the API with the filter structure provided in RESTful.  Doesnt seem to be a lot of people using the filter['name']=stuff structure.  

However the curl idea is fairly interesting as a solution.  Although I'd rather just use the API as it is.  Is it possible to change the structure of how filters are sent to the api?  Maybe I can just roll something custom like filternamehere=fileterby or something.

**Comment by e0ipso** [comment]
> Doesnt seem to be a lot of people using the filter['name']=stuff structure.

There's a lot of people using it. For one, I have been using it for more than a year now.

> However the curl idea is fairly interesting as a solution. Although I'd rather just use the API as it is.

You will not need any modification to the API, cURL is just a very common command line utility to make HTTP requests. I was asking about that because I don't understand what you are trying to get (and how) from your API. Having it expressed in another format may help you (and others) to understand what's going on.
